src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -106,19 +111,41 @@ protected void recordLineage(LineageRecorder lineageRecorder, List<String> outpu
 
     @Override
     public void validate() {
-      // TODO: CDAP-15900 add failure collector
-      super.validate(null);
+      // no-op
+    }
+
+    @Override
+    public void validate(FailureCollector collector) {
+      super.validate(collector);
       // validate that path is valid
-      if (!containsMacro("path")) {
-        GCSPath.from(path);
+      if (!containsMacro(NAME_PATH)) {
+        try {
+          GCSPath.from(path);
+        } catch (IllegalArgumentException e) {
+          collector.addFailure(e.getMessage(), null).withConfigProperty(NAME_PATH).withStacktrace(e.getStackTrace());
+        }
       }
-      if (suffix != null && !containsMacro("suffix")) {
-        new SimpleDateFormat(suffix);
+      if (suffix != null && !containsMacro(NAME_SUFFIX)) {
+        try {
+          new SimpleDateFormat(suffix);
+        } catch (IllegalArgumentException e) {
+          collector.addFailure("Invalid suffix.", "Ensure provided suffix is valid.")
```

Comment:
should we include the message from the exception? Otherwise it won't be apparent why it's invalid.

Also don't think we need the corrective action if we include the exception message